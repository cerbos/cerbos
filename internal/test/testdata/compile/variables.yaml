---
mainDef: resource_policies/example.yaml
inputDefs:
  resource_policies/example.yaml:
    apiVersion: api.cerbos.dev/v1
    variables:
      foo: "1"
    resourcePolicy:
      resource: example
      version: default
      variables:
        import:
          - example
        local:
          qux: "3"
  export_variables/example.yaml:
    apiVersion: api.cerbos.dev/v1
    exportVariables:
      name: example
      definitions:
        bar: "2"
