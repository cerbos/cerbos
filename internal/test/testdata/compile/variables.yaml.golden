{
  "fqn": "cerbos.resource.example.vdefault",
  "resourcePolicy": {
    "meta": {
      "fqn": "cerbos.resource.example.vdefault",
      "resource": "example",
      "version": "default"
    },
    "policies": [
      {
        "variables": {
          "bar": {
            "original": "2",
            "checked": {
              "typeMap": {
                "1": {
                  "primitive": "INT64"
                }
              },
              "sourceInfo": {
                "location": "<input>",
                "lineOffsets": [
                  2
                ],
                "positions": {
                  "1": 0
                }
              },
              "expr": {
                "id": "1",
                "constExpr": {
                  "int64Value": "2"
                }
              }
            }
          },
          "foo": {
            "original": "1",
            "checked": {
              "typeMap": {
                "1": {
                  "primitive": "INT64"
                }
              },
              "sourceInfo": {
                "location": "<input>",
                "lineOffsets": [
                  2
                ],
                "positions": {
                  "1": 0
                }
              },
              "expr": {
                "id": "1",
                "constExpr": {
                  "int64Value": "1"
                }
              }
            }
          },
          "qux": {
            "original": "3",
            "checked": {
              "typeMap": {
                "1": {
                  "primitive": "INT64"
                }
              },
              "sourceInfo": {
                "location": "<input>",
                "lineOffsets": [
                  2
                ],
                "positions": {
                  "1": 0
                }
              },
              "expr": {
                "id": "1",
                "constExpr": {
                  "int64Value": "3"
                }
              }
            }
          }
        }
      }
    ]
  }
}